src/keys/KeyManager.ts

```diff
@@ -502,16 +558,35 @@ class KeyManager {
   protected async setupRootKeyPair(
     password: string,
     bits: number = 4096,
-  ): Promise<KeyPair> {
-    let rootKeyPair;
+    recoveryCode: string | undefined,
+  ): Promise<[KeyPair, string]> {
+    let rootKeyPair: KeyPair;
+    let newRecoveryCode: string;
+    const recoveryCode_ = recoveryCode ?? keysUtils.generateRecoveryCode();
     if (await this.existsRootKeyPair()) {
-      rootKeyPair = await this.readRootKeyPair(password);
+      if (recoveryCode != null) {
+        // Recovery code is provided, we want to reset the password.
+        this.logger.info('Resetting root key pair password');
+        rootKeyPair = await KeyManager.recoverKeyManager_(
+          recoveryCode,
+          this.keysPath,
+          password,
+          bits,
+          this.fs,
+        );
+        newRecoveryCode = recoveryCode;
+      } else {
+        // Recovery code was not provided, read the root key as normal.
+        rootKeyPair = await this.readRootKeyPair(password);
+        newRecoveryCode = '';
+      }
     } else {
       this.logger.info('Generating root key pair');
-      rootKeyPair = await this.generateKeyPair(bits);
+      rootKeyPair = await this.generateKeyPair(bits, recoveryCode_);
       await this.writeRootKeyPair(rootKeyPair, password);
+      newRecoveryCode = recoveryCode_;
     }
-    return rootKeyPair;
```

Comment:
This should just be `recoveryCode` or `recoveryCode_`.

Reply:
It just returns the `rootKeyPair` now. 

This was hard to find, in future when commenting on code here you can select a range of lines to comment on. This way you can include the full changes or relevant lines to comment on.